apps/api/src/app/events/usecases/trigger-event/message-filter.matcher.ts

```diff
@@ -1,17 +1,28 @@
-import { NotificationStepEntity, SubscriberEntity } from '@novu/dal';
+import { NotificationStepEntity, SubscriberEntity, EnvironmentRepository, SubscriberRepository } from '@novu/dal';
 import { ITriggerPayload } from '@novu/node';
 import * as _ from 'lodash';
 import got from 'got';
+import { createHmac } from 'crypto';
+import { CreateExecutionDetails } from '../../../execution-details/usecases/create-execution-details/create-execution-details.usecase';
+import { SendMessageCommand } from '../send-message/send-message.command';
 
 export interface IFilterVariables {
   payload: ITriggerPayload;
   subscriber?: SubscriberEntity;
   webhook?: Record<string, unknown>;
 }
 
+export interface IMessageFilterConfiguration {
+  command: SendMessageCommand;
+  subscriberRepository: SubscriberRepository;
+  createExecutionDetails: CreateExecutionDetails;
+  environmentRepository: EnvironmentRepository;
+}
+
 export async function matchMessageWithFilters(
```

Comment:
Would suggest to refactor this for a class `MessageFilter` and than we can persist some mandatory and reused variables in the constructor, and class members could access it instead of doing the prop drilling on all the functions.

Reply:
https://github.com/novuhq/novu/pull/2348/commits/be263ad7459d4ccd43accc2b7234080d3af24482